lib/Doctrine/DBAL/Sharding/PoolingShardConnection.php

```diff
@@ -69,34 +70,34 @@ class PoolingShardConnection extends Connection
     public function __construct(array $params, Driver $driver, ?Configuration $config = null, ?EventManager $eventManager = null)
     {
         if (! isset($params['global'], $params['shards'])) {
-            throw new InvalidArgumentException("Connection Parameters require 'global' and 'shards' configurations.");
+            throw new InvalidArgumentException('Connection Parameters require "global" and "shards" configurations.');
         }
 
         if (! isset($params['shardChoser'])) {
-            throw new InvalidArgumentException("Missing Shard Choser configuration 'shardChoser'");
+            throw new InvalidArgumentException('Missing Shard Choser configuration "shardChoser".');
         }
 
         if (is_string($params['shardChoser'])) {
             $params['shardChoser'] = new $params['shardChoser']();
         }
 
         if (! ($params['shardChoser'] instanceof ShardChoser)) {
-            throw new InvalidArgumentException("The 'shardChoser' configuration is not a valid instance of Doctrine\DBAL\Sharding\ShardChoser\ShardChoser");
+            throw new InvalidArgumentException('The "shardChoser" configuration is not a valid instance of Doctrine\DBAL\Sharding\ShardChoser\ShardChoser');
         }
 
         $this->connectionParameters[0] = array_merge($params, $params['global']);
 
         foreach ($params['shards'] as $shard) {
             if (! isset($shard['id'])) {
-                throw new InvalidArgumentException("Missing 'id' for one configured shard. Please specify a unique shard-id.");
+                throw new InvalidArgumentException('Missing "id" for one configured shard. Please specify a unique shard-id.');
             }
 
             if (! is_numeric($shard['id']) || $shard['id'] < 1) {
                 throw new InvalidArgumentException('Shard Id has to be a non-negative number.');
             }
 
             if (isset($this->connectionParameters[$shard['id']])) {
-                throw new InvalidArgumentException('Shard ' . $shard['id'] . ' is duplicated in the configuration.');
+                throw new InvalidArgumentException(sprintf('Shard "%s" is duplicated in the configuration.', $shard['id']));
```

Comment:
Is `$shard['id']` a string?

Reply:
It's annotated as `int|string` in most places but looks like it's a numeric string (or an integer):

https://github.com/doctrine/dbal/blob/82f5fe317bedf9f328c06aeaab5a2f586810e94a/lib/Doctrine/DBAL/Sharding/PoolingShardConnection.php#L94-L96

Also, the `id` comes from `sys.federation_member_distributions.member_id`:

https://github.com/doctrine/dbal/blob/68e48e83021c22f19354558a5fdb46e53e9e6c1a/lib/Doctrine/DBAL/Sharding/SQLAzure/SQLAzureShardManager.php#L158-L164

which according to this [article](https://convective.wordpress.com/2012/03/05/introduction-to-sql-azure-federations/) is `INT`.

These details are mostly for myself, I'll need this info later to finish the work on strict types.